rclcpp/include/rclcpp/wait_set_policies/dynamic_storage.hpp

```diff
@@ -216,6 +216,11 @@ class DynamicStorage : public rclcpp::wait_set_policies::detail::StoragePolicyCo
       shared_waitables_
     );
 
+    if(this->needs_pruning_) {
```

Comment:
Again, this is weird to me that it passed the linters, I think they may be broken. I'll look into that, but it shouldn't block this pr, we can fix it up in a subsequent pr if that's the case I guess.

Reply:
Hm, I use the ament_uncrustify and ament_cpplint things, like I use clang_format.
Meaning, I just write code using 'my' coding style and let the auto formaters do their thing later on, and don't look at the format any more after the uncrustify // cpplint. 
Is there a better tool for my workflow, that does not produce these strange artifacts ?

Reply:
that is the same with what i usually do, if there are any coding style problem, `colcon test` (or CI) should say something... i tried this my local environment, all green too. i am not sure what happening here, some rules are changed??? @ros2/team any thoughts?

Reply:
fixed